python/tests/catalog/integration_test_glue.py

```diff
@@ -0,0 +1,304 @@
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+
+import os
+
+import boto3
+import pytest
+from botocore.exceptions import ClientError
+
+from pyiceberg.catalog.glue import GlueCatalog
+from pyiceberg.exceptions import (
+    NamespaceAlreadyExistsError,
+    NamespaceNotEmptyError,
+    NoSuchNamespaceError,
+    NoSuchTableError,
+    TableAlreadyExistsError,
+)
+from pyiceberg.schema import Schema
+from tests.catalog.test_glue import (
+    get_random_database_name,
+    get_random_databases,
+    get_random_table_name,
+    get_random_tables,
+)
+
+# The number of random characters in generated table/database name
+RANDOM_LENGTH = 20
+# The number of tables/databases used in list_table/namespace test
+LIST_TEST_NUMBER = 2
+
+
+def get_bucket_name():
```

Comment:
> We can keep these tests and make them more formal integration tests, maybe use a flag `AWS_INTEGRATION_TEST_ENABLED` and if true then don't skip these tests, and get proper configurations of a AWS environment.

[Update] I moved and formalized all integration tests in to `integration_test_glue.py` so it will not be collected by `pytest` automatically.  Users can run integration test manually by setting an environment variable `AWS_TEST_BUCKET` and account credentials.